ts/screens/profile/ShareDataScreen.tsx

```diff
@@ -64,8 +68,17 @@ const ShareDataScreen = (props: Props): React.ReactElement => {
             "profile.main.privacy.shareData.screen.cta.dontShareData"
           ),
           onPress: () => {
-            trackMixpanelSetEnabled(true, getFlowType(false, false));
+            // Before tracking any event, we need to enable mixpanel
             props.setMixpanelEnabled(true);
+            // We wait some time to allow mixpanel to be enabled
+            // before tracking the event
+            setTimeout(() => {
+              void trackMixpanelSetEnabled(
+                true,
+                getFlowType(false, false),
+                store.getState()
+              );
+            }, 1000);
             IOToast.success(
```

Comment:
Since we are in a functional component, you can use the `useIOToast` hook instead of `IOToast` ref. What do you think?

Reply:
Yup! Right! But I prefer to leave this PR relative to the MP event only. We'll take care of a more deep refactoring of the page when we approach [this.]([IOPID-1506](https://pagopa.atlassian.net/browse/IOPID-1506)). What do you think?

Reply:
It makes sense!